src/metrics/mod.rs

```diff
@@ -51,12 +51,15 @@ impl Actor for MetricsService {
             .listener
             .try_clone()
             .expect("failed to clone metrics listener");
+        listener
+            .set_nonblocking(true)
```

Comment:
This addresses https://github.com/coreos/zincati/pull/536#issuecomment-825335567.
In `tokio 0.3.2`, the `tokio::net::UnixListener::from_std()` method no longer helps us set Unix listeners to nonblocking, so we need to do it ourselves now.